pkg/trace/api/otlp_test.go

```diff
@@ -2323,6 +2324,16 @@ func TestMarshalEvents(t *testing.T) {
 					"dropped_attributes_count":2
 				}]`,
 		},
+		{
+			in: makeEventsSlice(`something:"nested"`, map[string]string{
+				`abc"def"`: `123"456"`,
```

Comment:
nit: Can you also test that arrays are encoded correctly? I'm worried that the following input would be malformed:

```
`str_array`: [`test"1`, `"test2`]
```